include/wlr/types/wlr_keyboard.h

```diff
@@ -70,9 +70,13 @@ struct wlr_event_keyboard_key {
 	enum wlr_key_state state;
 };
 
+struct wlr_input_device;
+
+struct wlr_keyboard *wlr_keyboard_create(struct wlr_input_device *dev);
```

Comment:
I'm not sure how this could work. Currently, the `wlr_keyboard` is created by the backend, and then the backend calls `wlr_keyboard_send_{key,modifiers}` to sned events to the `wlr_keyboard`.

The `wlr_input_device` contains the `wlr_keyboard_interface` and we need a way to ask the backend to send key events to a `wlr_keyboard`. Maybe `wlr_keyboard_interface.attach(wlr_keyboard *kb)`?

Reply:
Slow down. The user should not be creating wlr keyboards, the backend should still be doing that. The approach here is wrong.

Reply:
Okay, I wanted to do this part:

>When a device is added, rootston creates a wlr-keyboard from the device and configures it based on compositor policy.

Reply:
I see. That's probably not the best design. We should still have the backend create a wlr_keyboard and push it up to rootston. The full quote is:

>We should transfer ownership of the wlr-keyboard to rootston. When a device is added, rootston creates a wlr-keyboard from the device and configures it based on compositor policy.

Ownership doesn't speak to who created it (wlr seat isn't creating it now, either), but rather who *manages* it.

Reply:
> Slow down. The user should not be creating wlr keyboards, the backend should still be doing that. The approach here is wrong.

Actually this is exactly what I had in mind. The wlr_keyboard is a wrapper around a device made by the user.

Reply:
I see. -1 to that design.

Reply:
Ok everybody stop working on this and let me do a more specific proposal.

Reply:
I don't think this is really that complicated. Your design was mostly on the money.